### PR TITLE
Port to scala.js 1.0 and scalajs-react 1.7.0

### PR DESCRIPTION
**Pull request by cquiroz**
This is a work in progress
Pending:
* Use public versions of diode and boopickle
* Eliminate deprecated calls
* Testing

### CODE REVIEW

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/ConnectionState.scala [hunk 1]
```diff
@@ -34,7 +33,7 @@ object ConnectionState {
       f"${delay / 1000}%d"
     }
 
-  private val component = ScalaComponent
+  val component = ScalaComponent
```
Comment (rpiaggio) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/ConnectionState.scala:36]: Just for future reference, we can keep these `protected`.

File: project/Settings.scala [hunk 2]
```diff
@@ -120,6 +121,12 @@ object Settings {
       "net.logstash.logback" % "logstash-logback-encoder" % LibraryVersions.logstash
     )
     val Log4s = Def.setting("org.log4s" %%% "log4s" % LibraryVersions.log4s)
+    val Log4CatsLogLevel = Def.setting(
+      Seq(
+        "io.chrisdavenport" %%% "log4cats-core"     % LibraryVersions.log4cats,
+        "com.rpiaggio"      %%% "log4cats-loglevel" % LibraryVersions.log4catsLevel
```
Comment (rpiaggio) [inline project/Settings.scala:127]: I think we are not using this one.

File: project/plugins.sbt [hunk 3]
```diff
@@ -4,7 +4,7 @@ resolvers  ++= Seq(
 )
 // Gives support for Scala.js compilation
 val scalaJSVersion =
-  Option(System.getenv("SCALAJS_VERSION")).getOrElse("0.6.33")
+  Option(System.getenv("SCALAJS_VERSION")).getOrElse("1.1.1")
```
Comment (rpiaggio) [inline project/plugins.sbt:7]: I think we are not using this.

**Comment by cquiroz** [comment]
This has been tested enough and we need to move forward to work on other fixes. I'll merge it and keep testing on master

**Pull request merged**